From 2de46c6e5743ca226e7e70399e379074b375fe49 Mon Sep 17 00:00:00 2001
From: Retrace CPython Builder <retrace-cpython@retracesoftware.com>
Date: Thu, 7 May 2026 09:50:59 +0100
Subject: [PATCH 2/6] Add _retrace coordinate module

---
 Modules/Setup.bootstrap.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Modules/Setup.bootstrap.in b/Modules/Setup.bootstrap.in
index e3e9b96..f579bee 100644
--- a/Modules/Setup.bootstrap.in
+++ b/Modules/Setup.bootstrap.in
@@ -30,6 +30,7 @@ _locale _localemodule.c
 _operator _operator.c
 _stat _stat.c
 _symtable symtablemodule.c
+_retrace retracemodule.c

 # for systems without $HOME env, used by site._getuserbase()
 @MODULE_PWD_TRUE@pwd pwdmodule.c
--
2.50.1 (Apple Git-155)
