From 71f1ca55dfc3c7c74aeb024f5d3e4a3f91b185ce Mon Sep 17 00:00:00 2001
From: nathan-retrace <nathan@retracesoftware.com>
Date: Tue, 5 May 2026 21:55:20 +0100
Subject: [PATCH 2/5] Add retrace instruction counter module

---
 Modules/Setup.bootstrap.in |   1 +

diff --git a/Modules/Setup.bootstrap.in b/Modules/Setup.bootstrap.in
index e3e9b96..f579bee 100644
--- a/Modules/Setup.bootstrap.in
+++ b/Modules/Setup.bootstrap.in
@@ -32,2 +32,3 @@ _stat _stat.c
 _symtable symtablemodule.c
+retrace retracemodule.c
 
