From cb5f6e7827836cc5cca6f537402a66fabcabb2ec Mon Sep 17 00:00:00 2001
From: Retrace CPython Builder <retrace-cpython@retracesoftware.com>
Date: Thu, 7 May 2026 09:50:59 +0100
Subject: [PATCH 4/6] Add retrace call_at callback

---
 Python/ceval.c   | 26 ++++++++++++++++++++++++--
 Python/pystate.c |  5 +++++
 2 files changed, 29 insertions(+), 2 deletions(-)

diff --git a/Python/ceval.c b/Python/ceval.c
index 3816a42..d90457d 100644
--- a/Python/ceval.c
+++ b/Python/ceval.c
@@ -20,7 +20,7 @@
 #include "pycore_pylifecycle.h"   // _PyErr_Print()
 #include "pycore_pymem.h"         // _PyMem_IsPtrFreed()
 #include "pycore_pystate.h"       // _PyInterpreterState_GET()
-#include "pycore_retrace.h"       // retrace thread helpers
+#include "pycore_retrace.h"       // retrace callbacks
 #include "pycore_sysmodule.h"     // _PySys_Audit()
 #include "pycore_tuple.h"         // _PyTuple_ITEMS()
 #include "pycore_emscripten_signal.h"  // _Py_CHECK_EMSCRIPTEN_SIGNALS
@@ -1299,6 +1299,24 @@ eval_frame_handle_pending(PyThreadState *tstate)
     #define USE_COMPUTED_GOTOS 0
 #endif
 
+#define RETRACE_CALL_AT() \
+    do { \
+        PyInterpreterState *retrace_interp = tstate->interp; \
+        if (((retrace_interp->retrace.call_at_armed && \
+              retrace_interp->retrace.call_at_thread_id == tstate->retrace.thread_id) || \
+             retrace_interp->retrace.call_at_extra_armed) && \
+            _PyRetrace_FrameIsVisible(frame)) \
+        { \
+            _PyFrame_SetStackPointer(frame, stack_pointer); \
+            int retrace_err = _PyRetrace_CheckCallAt(tstate, frame); \
+            stack_pointer = _PyFrame_GetStackPointer(frame); \
+            frame->stacktop = -1; \
+            if (retrace_err < 0) { \
+                goto error; \
+            } \
+        } \
+    } while (0)
+
 #ifdef Py_STATS
 #define INSTRUCTION_START(op) \
     do { \
@@ -1306,9 +1324,14 @@ eval_frame_handle_pending(PyThreadState *tstate)
         OPCODE_EXE_INC(op); \
         _py_stats.opcode_stats[lastopcode].pair_count[op]++; \
         lastopcode = op; \
+        RETRACE_CALL_AT(); \
     } while (0)
 #else
-#define INSTRUCTION_START(op) (frame->prev_instr = next_instr++)
+#define INSTRUCTION_START(op) \
+    do { \
+        frame->prev_instr = next_instr++; \
+        RETRACE_CALL_AT(); \
+    } while (0)
 #endif
 
 #if USE_COMPUTED_GOTOS
diff --git a/Python/pystate.c b/Python/pystate.c
index 1c93702..77f4084 100644
--- a/Python/pystate.c
+++ b/Python/pystate.c
@@ -413,4 +413,10 @@ interpreter_clear(PyInterpreterState *interp, PyThreadState *tstate)
     Py_CLEAR(interp->audit_hooks);
+    _PyRetrace_ClearSpaceCallbacks(interp);
+    _PyRetrace_ClearIdentityHashTable(interp);
+    interp->retrace.call_at_armed = 0;
+    Py_CLEAR(interp->retrace.call_at_coordinates);
+    Py_CLEAR(interp->retrace.call_at_callback);
+    Py_CLEAR(interp->retrace.call_at_overshoot_callback);
 
     PyConfig_Clear(&interp->config);
     Py_CLEAR(interp->codec_search_path);
--
2.50.1 (Apple Git-155)
