From dccbf162e3f0db77b9ae11a63231e1fa63395aad Mon Sep 17 00:00:00 2001
From: nathan-retrace <nathan@retracesoftware.com>
Date: Tue, 5 May 2026 21:57:15 +0100
Subject: [PATCH 5/6] Update frame size tests for retrace field

---
 Lib/test/test_sys.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Lib/test/test_sys.py b/Lib/test/test_sys.py
index 3f65d5e..b1d2354 100644
--- a/Lib/test/test_sys.py
+++ b/Lib/test/test_sys.py
@@ -1498,7 +1498,7 @@ class C(object): pass
         def func():
             return sys._getframe()
         x = func()
-        check(x, size('3Pi3c7P2ic??2P'))
+        check(x, size('3Pi3c7P2ic??4q2P'))
         # function
         def func(): pass
         check(func, size('14Pi'))
@@ -1515,7 +1515,7 @@ def bar(cls):
             check(bar, size('PP'))
         # generator
         def get_gen(): yield 1
-        check(get_gen(), size('P2P4P4c7P2ic??P'))
+        check(get_gen(), size('P2P4P4c7P2ic??4qP'))
         # iterator
         check(iter('abc'), size('lP'))
         # callable-iterator
--
2.50.1 (Apple Git-155)
