From 3c280860019125a2c655584c13c98bb06800c6bf Mon Sep 17 00:00:00 2001
From: nathan-retrace <nathan@retracesoftware.com>
Date: Tue, 5 May 2026 17:47:39 +0100
Subject: [PATCH 2/5] Add retrace instruction counter module

---
 Modules/Setup.bootstrap.in |   1 +

diff --git a/Modules/Setup.bootstrap.in b/Modules/Setup.bootstrap.in
index 8ef0f20..5224b0f 100644
--- a/Modules/Setup.bootstrap.in
+++ b/Modules/Setup.bootstrap.in
@@ -33,2 +33,3 @@ _stat _stat.c
 _symtable symtablemodule.c
+retrace retracemodule.c
 
