From 121d603a97c1f3d104c775f9132dbee0b3571208 Mon Sep 17 00:00:00 2001
From: nathan-retrace <nathan@retracesoftware.com>
Date: Tue, 5 May 2026 17:47:39 +0100
Subject: [PATCH 2/7] Add retrace instruction counter module

---
 Modules/Setup.bootstrap.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Modules/Setup.bootstrap.in b/Modules/Setup.bootstrap.in
index 8ef0f20..5224b0f 100644
--- a/Modules/Setup.bootstrap.in
+++ b/Modules/Setup.bootstrap.in
@@ -31,6 +31,7 @@ _locale _localemodule.c
 _operator _operator.c
 _stat _stat.c
 _symtable symtablemodule.c
+retrace retracemodule.c
 
 # for systems without $HOME env, used by site._getuserbase()
 @MODULE_PWD_TRUE@pwd pwdmodule.c
-- 
2.50.1 (Apple Git-155)

