From d8e774e083dc7f1b93bfb1c8d24d82e0ca011044 Mon Sep 17 00:00:00 2001
From: nathan-retrace <nathan@retracesoftware.com>
Date: Tue, 5 May 2026 20:49:25 +0100
Subject: [PATCH 5/5] Update frame size tests for retrace field

---
 Lib/test/test_sys.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Lib/test/test_sys.py b/Lib/test/test_sys.py
index 2f06299..5962572 100644
--- a/Lib/test/test_sys.py
+++ b/Lib/test/test_sys.py
@@ -1524,3 +1524,3 @@ def func():
         x = func()
-        check(x, size('3Pi3c7P2ic??2P'))
+        check(x, size('3Pi3c7P2ic??q2P'))
         # function
@@ -1541,3 +1541,3 @@ def bar(cls):
         def get_gen(): yield 1
-        check(get_gen(), size('PP4P4c7P2ic??2P'))
+        check(get_gen(), size('PP4P4c7P2ic??q2P'))
         # iterator
-- 
2.50.1 (Apple Git-155)

